docs/datagrid/datagrid-with-react.md

```diff
@@ -7,23 +7,57 @@ To create a DataGrid with Angular please follow the steps below: <br>
 
     ```bash
     npm install @highcharts/dashboards
-    ```
+    ````
 
-2. Import the Dashboards package.
+3. Create a DataGrid React component:
 
-    ```typescript
-    import * as Dashboards from '@highcharts/dashboards';
-    ```
+    ```tsx
+    // DataGrid.tsx
+
+    import { useEffect } from 'react';
+    import DG from '@highcharts/dashboards/datagrid';
+    import '@highcharts/dashboards/css/datagrid.css';
 
-3. Create a HTML container.  
+    export default function DataGrid(props: { config: DG.Options }) {
+        const { config } = props;
+
+        useEffect(() => {
+            DG.dataGrid('container', config);
+        }, [config]);
+
+        return (
+            <div id="container" />
+        );
+    }
 
-    Add a div where you want to render the dashboard:
-    ```html
-    <div id="container"></div>
     ```
 
-    You can refer to the element by its id or you can use the `ElementRef` to get the element.
+4. Use the component in your application:
+
+    ```tsx
+    // App.tsx
 
-4. Create a DataGrid using the factory function `DataGrid.dataGrid`. The function takes three arguments:
-    - `container` - the element where the DataGrid will be rendered, can be an id of the element or the direct reference to the element
-    - `options` - the options object for the DataGrid
+    import DataGrid from "./components/DataGrid";
+
+    function App() {
+
+        const config: DataGrid.Options = {
+            dataTable: {
+                columns: {
+                    name: ['Alice', 'Bob', 'Charlie', 'David'],
+                    age: [23, 34, 45, 56],
+                    city: ['New York', 'Oslo', 'Paris', 'Tokyo'],
+                }
+            }
+        }
+
+        return (
+            <div className="App">
+                <DataGrid config={config} />
```

Comment:
Should id (container) be sent as a prop instead of hardcoding it in the component? Component is not reusable as it stands now. This comments might also apply to angular and vue articles.

Reply:
This is how Highcharts integrations work, and it’s also the case in the Dashboards demos. The examples showcase a standalone implementation of the 'simplest' integration, which can then be extended further. If we get to the point where we implement official integrations, it will likely work similarly, where options can be passed as props to these components.

Reply:
Hm, OK. Just find it strange that we provide e.g. React component examples that are not reusable. Kind of the point with React components? But OK, get your point.

Reply:
What I don't like here is that in the case of React, we pass the container ID to the factory method, not its reference. It's the same in Dashboards, but I'll update it later, when doing the live demos.